### PR TITLE
Add wiki pages back (2nd try)

### PR DESCRIPTION
**Pull request by Tobi406**
This time, most things do actually work. Content is fetched from `/metadata/wiki/:page` which fetches from GitHub

## TODO:
- [x] Make the scroll listeners on _page.vue useable
- [x] Re-fetch every x hours

### CODE REVIEW

**Comment by Tobi406** [comment]
And I think I just shifted the problem...

**Comment by lucko** [comment]
:o would be cool if we could ditch having to use a submodule for this

**Comment by Tobi406** [comment]
Should work now. Still have to do some To-Do things but that is it then

**Comment by Tobi406** [comment]
And done now, finally :tada:

File: pages/wiki/_page.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+<script>
+const axios = require('axios');
+
+export default {
+  middleware ({route, redirect}) {
+    if (route.path === '/wiki') redirect('/wiki/Home');
+  },
+  async asyncData({ params }) {
+    const response = await axios.get(`http://localhost:3000/metadata/wiki/${params.page}`);
```
Comment (Tobi406) [inline pages/wiki/_page.vue:24]: Note that we have to change the url there later

**Comment by Tobi406** [comment]
I think I went a little bit crazy, I added

- a server middleware, so we can fetch our data from that instead [LuckPerms/metadata-api](https://github.com/LuckPerms/metadata-api)
- Verbose viewer works
- Tree viewer works

**Comment by Tobi406** [comment]
I'm gonna wait with the editor until #285 is merged though

File: components/ToolIntro.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        </div>
+        <template v-else-if="!error">
+          <a :href="`/${type}/demo`">
+            <button class="button demo-button">
+              View Demo
+            </button>
+          </a>
+          <p v-if="type === 'editor'">To start a new editor session, use one of the following commands:</p>
+          <p v-if="type === 'verbose'">To generate a verbose report, do the following in game or from console:</p>
+          <p v-if="type === 'treeview'">To generate a permissiont ree, do the following in game or from the console:</p>
```
Comment (Tobi406) [inline components/ToolIntro.vue:24]: Typo

File: pages/wiki/_page.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      } else {
+        document.getElementById('article').scrollTo({
+          top: 0,
+          behavior: 'smooth',
+        });
+      }
+    },
+  },
+}
+</script>
```
Comment (Tobi406) [inline pages/wiki/_page.vue:84]: Add newline

**Comment by Tobi406** [comment]
I think I just figured out serverMiddleware doesn't run on Netlify (because it isn't static), so I guess we still have a use for the other repo.

**Comment by lucko** [comment]
Could it be something that we add to the metadata API?

**Comment by Tobi406** [comment]
It's basicly just an express server (like on the metadata-api) running in nuxt. Nothing serious, can just be added to the metadata-api.

**Comment by Tobi406** [comment]
As I said already, Netlify only supports static pages. 
We can generate everything static, with the exception of the editor etc.
I think we do have the following options right now, feel free to correct me:
- No longer use Netlify and use `npm run build && npm run start` to run it ourselves
- Just leave this Nuxt thing
- *not sure about this:* Change the URLs back to using a query or hash.
  - Would break versions using the old URL, even tough we might maybe be able to make a redirect *(not sure again)*

File: package.json [hunk 4]
```diff
@@ -6,20 +6,26 @@
   "private": true,
   "scripts": {
     "dev": "nuxt",
-    "build": "nuxt build",
+    "build": "nuxt build --target static && nuxt export",
```
Comment (Turbotailz) [inline package.json:9]: What is this doing?

Comment (Tobi406) [reply package.json:9]: I think it does the same as `nuxt generate`. Was more for me to test. Also see my [previous comment](https://github.com/lucko/LuckPermsWeb/pull/283#issuecomment-651070482).

**Comment by Turbotailz** [comment]
Sorry I've been silent on this, kinda been lurking as you've made updates.

Overall I think you've done a great job so far and thanks for taking the initiative 💪

I would love to continue using Netlify if possible, because then I can still deploy updates to production without needing to nag Luck about it, plus they handle all the annoying stuff like bandwidth, caching, etc. I haven't really had much chance to play around with Nuxt to really get a good idea on what is the best solution here, though.

Personally not a big fan of the HTTP requests for getting the wiki contents but submodules aren't really a great solution either. Can't really add much input at this stage. Would definitely prefer to continue using the metadata API, however. If the wiki does need to be queried via HTTP then we can add it to that instead.

> Change the URLs back to using a query or hash

Definitely not keen on this haha.

**Pull request closed without merging**